### ISSUE TITLE
Support Camera Calibration?

### ISSUE THREAD

**Issue by kingsjl** [openmv/openmv#572]
Support Camera Calibration?
https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_calib3d/py_calibration/py_calibration.html

**Comment by iabdalkader** [comment]
Hi, you can calibrate the camera by saving images of the patterns using the lens you want on the SD card, and then running calibrate.py on PC.

**Comment by chobitsfan** [comment]
Hi @kwagyeman Is it possible to add a remap function that accept a map of (x,y) points (or map1 for x, map2 for y)? So that usrs can calibrate lens and use initUndistortRectifyMap() to calculate map on host PC and send map array to openmv. It would be very helpful for improving template matching result. Thank you very much

**Comment by kwagyeman** [comment]
@chobitsfan Yes, it's possible. However, the only easy thing to do is with the nearest neighbor, which will look quite ugly. The N6's GPU can do this, though, for us, in hardware.

ST wrote this: https://github.com/STMicroelectronics/stm32-mw-ipl/blob/main/Src/stm32ipl_dewarp.c

(They lifted our library to write their library - stuff there is out of date now.)

However, those are software-only; performance will not be great. But the N6's GPU can do a texture remap while handling bilinear scaling.

What processor do you need to do this for? If it's the N6... I can schedule some work on this in a few months. Doing the pure software mode support is of dubious value.

**Comment by chobitsfan** [comment]
Hi @kwagyeman Thank you for replying. 

> What processor do you need to do this for? If it's the N6... I can schedule some work on this in a few months. Doing the pure software mode support is of dubious value.

I am sorry but I am using AE3. So I think I would need to leave undistort job to connected Raspberry Pi?

**Comment by kwagyeman** [comment]
So... like, to do a good job fast here, I have to write all of this with ARM Helium. This is a similar level of effort to our image debayering code, which took a solid month of programming effort. This may be faster to do now with AI, but, it's still a huge amount of work.

If you want this feature, you'll want to jump into the C code and build it so that it gets done faster.

**Comment by chobitsfan** [comment]
Hi @kwagyeman Thank you so much for pointing me in the right direction.
